src/libraries/System.Text.RegularExpressions/tests/RegexReductionTests.cs

```diff
@@ -535,70 +557,100 @@ public void PatternsReduceDifferently(string pattern1, string pattern2)
+        [InlineData(@"a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(a(ab|cd+)|ef+)|gh+)|ij+)|kl+)|mn+)|op+)|qr+)|st+)|uv+)|wx+)|yz+)|01+)|23+)|45+)|67+)|89+)|AB+)|CD+)|EF+)|GH+)|IJ+)|KL+)|MN+)|OP+)|QR+)|ST+)|UV+)|WX+)|YZ+)", RegexOptions.None, 3, null)]
+        [InlineData(@"(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((a)))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))", RegexOptions.None, 1, 1)]
+        [InlineData(@"(?(\d)\d{3}|\d)", RegexOptions.None, 1, 3)]
+        [InlineData(@"(?(\d{7})\d{3}|\d{2})", RegexOptions.None, 2, 3)]
+        [InlineData(@"(?(\d{7})\d{2}|\d{3})", RegexOptions.None, 2, 3)]
+        [InlineData(@"(?(\d)\d{3}|\d{2})", RegexOptions.None, 2, 3)]
+        [InlineData(@"(?(\d)|\d{2})", RegexOptions.None, 0, 2)]
+        [InlineData(@"(?(\d)\d{3})", RegexOptions.None, 0, 3)]
+        [InlineData(@"(abc)(?(1)\d{3}|\d{2})", RegexOptions.None, 5, 6)]
+        [InlineData(@"(abc)(?(1)\d{2}|\d{3})", RegexOptions.None, 5, 6)]
+        [InlineData(@"(abc)(?(1)|\d{2})", RegexOptions.None, 3, 5)]
+        [InlineData(@"(abc)(?(1)\d{3})", RegexOptions.None, 3, 6)]
+        [InlineData(@"(abc|)", RegexOptions.None, 0, 3)]
+        [InlineData(@"(|abc)", RegexOptions.None, 0, 3)]
+        [InlineData(@"(?(x)abc|)", RegexOptions.None, 0, 3)]
+        [InlineData(@"(?(x)|abc)", RegexOptions.None, 0, 3)]
+        [InlineData(@"(?(x)|abc)^\A\G\z\Z$", RegexOptions.None, 0, 3)]
+        [InlineData(@"(?(x)|abc)^\A\G\z$\Z", RegexOptions.Multiline, 0, 3)]
+        [InlineData(@"^\A\Gabc", RegexOptions.None, 3, null)] // leading anchor currently prevents ComputeMaxLength from being invoked, as it's not needed
+        [InlineData(@"^\A\Gabc", RegexOptions.Multiline, 3, null)]
+        [InlineData(@"abc            def", RegexOptions.IgnorePatternWhitespace, 6, 6)]
+        [InlineData(@"abcdef", RegexOptions.RightToLeft, 6, null)]
+        public void MinMaxLengthIsCorrect(string pattern, RegexOptions options, int expectedMin, int? expectedMax)
         {
-            var r = new Regex(pattern);
-            Assert.Equal(expectedLength, GetMinRequiredLength(r));
+            var r = new Regex(pattern, options);
+            Assert.Equal(expectedMin, GetMinRequiredLength(r));
+            if (!pattern.EndsWith("$", StringComparison.Ordinal) &&
+                !pattern.EndsWith(@"\Z", StringComparison.OrdinalIgnoreCase))
+            {
+                // MaxPossibleLength is currently only computed/stored if there's a trailing End{Z} anchor as the max length is otherwise unused
+                r = new Regex($"(?:{pattern})$", options);
+            }
+            Assert.Equal(expectedMax, GetMaxPossibleLength(r));
```

Comment:
NIT: Should we also check the FindOptimization's value for the Node? This will add some coverage to the case were we expect `FindOptimization.TrailingAnchor_FixedLength` on cases where min and max are the same and the trailing anchor was found.

Reply:
I think it'd be reasonable to add a separate test that validates the FindMode is what we expect it to be for a given pattern, but I think that's separate from this PR.